guiapp/src/components/InputWithUnits.vue

```diff
@@ -0,0 +1,251 @@
+                  {{scope.opt}}
+                </q-item-section>
+                <q-item-section side>
+                  {{formatNumber(evalString(scope.opt),1)}}
+                </q-item-section>
+              </q-item>
+            </template>
+          </q-select>
+        </div>
+        <div
+            class="side_note text-grey-9 q-px-xs"
+            style="width: 3rem"
+        >
+          {{units}}
+        </div>
+      </q-card-section>
+    </q-card>
+  </div>
+</template>
+
+<script lang="ts">
+import {evaluate} from 'mathjs';
+import {
+  defineComponent,
+  ref,
+  watch,
+  } from 'vue';
+
+export default defineComponent({
+  name: 'InputWithUnits',
+  props: {
+    inputResult: {
+      type: Number,
+      required: true,
+      default: 0
+    },
+    initialExpression: {
+      type: String,
+      required: true,
+      default: ''
+    },
+    title: {
+      type: String,
+      default: ''
+    },
+    units: {
+      type: String,
+      default: ''
+    },
+    isShowingHelp: {
+      type: Boolean,
+      default: false
+    }
+  },
+  emits: [
+    'update:input-result',
+    'update:is-showing-help'
+  ],
+  setup(props, {emit}) {
+
+    let localQSelectValue = ref('')
+    let localTooltipText = ref('')
+    let isShowingTooltip = ref(false)
+    let isShowingTooltipAppend = ref(false)
+    let isShowingHelpLocal = ref(false)
+
+    let evaluatedValue = ref(0)
+    let count_updates = 0
+
+    // Set some random values to get correct typing with
+    // TypeScript and remove them after initialization.
+    let qSelectOptions = ref(['a','b'])
+    let qSelectOptionsHistory = ref(['a','b'])
+    qSelectOptions.value.pop()
+    qSelectOptions.value.pop()
+    qSelectOptionsHistory.value.pop()
+    qSelectOptionsHistory.value.pop()
+
+    function filterQSelectOptions (val:string,
+                       update:(data: ()=>void) => void) {
+      update(() => {
+        // To remove the selection from previously
+        // selected option - we refill the options list
+        qSelectOptions.value = qSelectOptionsHistory.value
+      })
+    }
+
+    function formatNumber (value:string, digits:number):string {
+      if (value==='') return ''
+      // const help_string = 'text\n text'
+      const num = parseFloat(value)
+      if ( num < Math.pow(10, -digits) ||
+          num > 5*Math.pow(10,  digits+2)
+      ) return '='+num.toExponential(digits)
+      return '='+Number(Math.round(
+              parseFloat(value + 'e' + digits.toString())).toString()
+          + 'e-' + digits.toString()).toString()
+    }
+
+    // evaluate option items, returns empty string for trivial evaluation or error
+    function evalString(val:string):string {
+      // Using try{} block to drop silently invalid input
+      try {
+        const tryEvaluate = Number(evaluate(val))
+        if (!isNaN(tryEvaluate) && tryEvaluate != Number(val)) {
+          return tryEvaluate.toString()
+        }
+      } catch { }
```

Comment:
Catch what?

Reply:
To drop silently invalid input we need an empty catch block.